pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -345,8 +410,12 @@ let
       FONT_TER16x32 = whenAtLeast "5.0" yes;
     };
 
-    video = {
+    video = let
+      whenHasDevicePrivate = mkIf (!stdenv.isx86_32 && versionAtLeast version "5.1");
+    in {
       DRM_LEGACY = whenOlder "6.8" no;
+      DRM_SIMPLEDRM = yes;
```

Comment:
There is some discussion in https://github.com/NixOS/nixpkgs/issues/155533, in particular it might conflict with `FB_SIMPLE` somehow?

Reply:
Haven't re-read the thread now, but last time I failed to understand whether this will be an actual problem for certain hardware: https://github.com/NixOS/nixpkgs/issues/155533#issuecomment-1848598448.